src/zenml/integrations/kubernetes/orchestrators/kubernetes_orchestrator.py

```diff
@@ -785,6 +785,253 @@ def _stop_run(
+        else:
+            # Pod is missing - check events to determine if this indicates failure or unknown status
+            logger.warning(
+                f"Pod {pod_name} not found in namespace "
+                f"{self.config.kubernetes_namespace}"
+            )
+
+            indicates_failure = self._pod_has_failure_events(pod_name)
+
+            if indicates_failure:
+                # We have evidence of actual failure/deletion
+                logger.warning(
+                    f"Pod {pod_name} failed based on events. "
+                    f"Marking as failed."
+                )
+                return kube_utils.PodPhase.FAILED
+            else:
+                # No clear evidence of failure - could be cleanup, TTL, etc.
+                logger.warning(
+                    f"Pod {pod_name} not found but no clear failure events. "
+                    f"Marking as unknown."
+                )
+                return kube_utils.PodPhase.UNKNOWN
+
+    def _map_pod_phase_to_execution_status(
+        self, pod_phase: kube_utils.PodPhase
+    ) -> ExecutionStatus:
+        """Map Kubernetes pod phase to ZenML execution status.
+
+        Args:
+            pod_phase: The Kubernetes pod phase.
+
+        Returns:
+            The corresponding ZenML execution status.
+        """
+        if pod_phase == kube_utils.PodPhase.PENDING:
+            return ExecutionStatus.INITIALIZING
+        elif pod_phase == kube_utils.PodPhase.RUNNING:
+            return ExecutionStatus.RUNNING
+        elif pod_phase == kube_utils.PodPhase.SUCCEEDED:
+            return ExecutionStatus.COMPLETED
+        elif pod_phase == kube_utils.PodPhase.FAILED:
+            return ExecutionStatus.FAILED
+        else:  # UNKNOWN
+            return ExecutionStatus.FAILED
+
+    def _map_job_status_to_execution_status(
+        self, job: k8s_client.V1Job
+    ) -> Optional[ExecutionStatus]:
+        """Map Kubernetes job status to ZenML execution status.
+
+        Args:
+            job: The Kubernetes job.
+
+        Returns:
+            The corresponding ZenML execution status, or None if no clear status.
+        """
+        # Check job conditions first
+        if job.status and job.status.conditions:
+            for condition in job.status.conditions:
+                if condition.type == "Complete" and condition.status == "True":
+                    return ExecutionStatus.COMPLETED
+                elif condition.type == "Failed" and condition.status == "True":
+                    return ExecutionStatus.FAILED
+
+        # Check job status fields
+        if job.status:
+            if job.status.active and job.status.active > 0:
+                return ExecutionStatus.RUNNING
+            elif job.status.succeeded and job.status.succeeded > 0:
+                return ExecutionStatus.COMPLETED
+            elif job.status.failed and job.status.failed > 0:
+                return ExecutionStatus.FAILED
```

Comment:
This is wrong, as the job will retried failures. I think in general, just checking the conditions of the job should be fine, no need to also check the status. The job might also be running even if active is `0`, if it's in the process of spinning up a new pod.

Reply:
Removed status-based checks as discussed.